dual_net.py

```diff
@@ -173,10 +199,137 @@ def train(self, boards, pi, z):
         should match input_shape and action_size as set during initialization.
         returns the batch loss
         """
+        self.boards = [state_to_board(board) for board in boards]
         self.sess.run([self.update_op], feed_dict={self.board_placeholder: boards,
                                                    self.pi: pi,
                                                    self.z: z})
         loss = self.sess.run([self.loss], feed_dict={self.board_placeholder: boards,
                                                      self.pi: pi,
                                                      self.z: z})
         return loss
+
+# Map from piece to layer in net input.  0-5 are white.
+INDEX_TO_PIECE_MAP = {0: chess.KING, 6: chess.KING,
+                      1: chess.QUEEN, 7: chess.QUEEN,
+                      2: chess.ROOK, 8: chess.ROOK,
+                      3: chess.BISHOP, 9: chess.BISHOP,
+                      4: chess.KNIGHT, 10: chess.KNIGHT,
+                      5: chess.PAWN, 11: chess.PAWN}
+
+KQK_INDEX_TO_PIECE_MAP = {0: chess.KING,
+                          1: chess.QUEEN,
+                          2: chess.KING}
+
+CHAR_TO_INDEX_MAP = {'K': 0, 'k': 6,
+                     'Q': 1, 'q': 7,
+                     'R': 2, 'r': 8,
+                     'B': 3, 'b': 9,
+                     'N': 4, 'n': 10,
+                     'P': 5, 'p': 11}
+
+
+def state_to_board(state):
+    """
+    Transforms a state representation according to the full Chess board input
+    into its chess Board equivalent.
+    Parameters
+    ----------
+    state: a numpy object representing the input board
+
+    Returns a chess.Board object
+    """
+    pieces = {}
+    if state.shape[2] == 13:
+        for i in range(12):
+            piece = INDEX_TO_PIECE_MAP[i]
+            if i < 6:
+                color = chess.WHITE
+            else:
+                color = chess.BLACK
+            indices = np.argwhere(state[:, :, i] == 1)
+            squares = []
+            for coords in indices:
+                x, y = coords
+                squares.append(8 * (7 - x) + y)
+            for square in squares:
+                pieces[square] = chess.Piece(piece, color)
+        board = chess.Board()
+        board.set_piece_map(pieces)
+        board.turn = int(state[0, 0, 12])
+        return board
+    else:
```

Comment:
Seems like this has some overlap with the KQK ChessEnv in the other PR. Is our intention to merge them?